excludes clients with usermode +k / +X or the privs to set them from netrider kick

diff -r 41ce42398a8f ircd/m_burst.c
--- a/ircd/m_burst.c	Sun Jan 11 18:08:49 2009 +0000
+++ b/ircd/m_burst.c	Sun Jan 11 22:38:38 2009 +0000
@@ -299,6 +299,12 @@
           nmember = member->next_member;
           if (!MyUser(member->user) || IsZombie(member))
             continue;
+          /* Do not kick +k user or operator with priv to set it */
+          if (IsChannelService(member->user) || HasPriv(member->user, PRIV_CHANSERV))
+            continue;
+          /* Do not kick +X user or operator with priv to set it */
+          if (IsXtraOp(member->user) || HasPriv(member->user, PRIV_XTRA_OPER))
+            continue;
           /* Kick as netrider if key mismatch *or* remote channel is
            * +i (unless user is an oper) *or* remote channel is +r
            * (unless user has an account).
